lib/timeout.rb

```diff
@@ -132,6 +132,12 @@ def self.ensure_timeout_thread_created
         unless @timeout_thread and @timeout_thread.alive?
           @timeout_thread = create_timeout_thread
         end
+
+        # shut down timeout thread at exit
+        Kernel.at_exit do
+          @timeout_thread.kill
+          @timeout_thread.join rescue nil
```

Comment:
I'd prefer without the `rescue null` so exceptions are not swallowed silently.

Reply:
Shutdown of the timeout thread will be happening at exit and should not make more noise than leaving the thread running and letting it die.

Reply:
Additionally, `at_exit` behavior would dictate that a failed hook replaces a successful shutdown with an error code, so not swallowing this error would cause a visible behavior change versus allowing the thread to silently be killed (which is what the runtime does internally, albeit at the last possible moment).